xbmc/cores/paplayer/AudioDecoder.cpp

```diff
@@ -58,7 +58,10 @@ void CAudioDecoder::Destroy()
   m_pcmBuffer.Destroy();
 
   if ( m_codec )
+  {
+    m_codec->DeInit();
```

Comment:
Since the `DeInit` is set in` ICodec` I am assuming that it will be called from the outside and not in the child class itself uses the `ICodec`.